src/azure-cli/azure/cli/command_modules/monitor/_params.py

```diff
@@ -381,6 +381,20 @@ def load_arguments(self, _):
         c.argument('retention_in_days', options_list='--retention-time', type=int, required=True)
     # endregion
 
+    # region Log Analytics Workspace Data Export
+    with self.argument_context('monitor log-analytics workspace data-export') as c:
+        c.argument('data_export_name', options_list=['--name', '-n'], help="Name of the data export rule")
+        c.argument('workspace_name', options_list='--workspace-name')
+        c.argument('enable_all_tables', options_list=['--all', '--enable-all-tables'], arg_type=get_three_state_flag(),
+                   help="All workspace's tables are exported when this is enabled.")
+        c.argument('table_names', nargs='+', options_list=['--tables', '-t'], help='An array of tables to export.')
```

Comment:
add help "if --export-all-tables is true, this argument should not be provided."